runtime/index.d.ts

```diff
@@ -35,3 +36,22 @@ declare module "@sapper/service-worker" {
 	export const shell: string[];
 	export const routes: Array<{ pattern: RegExp }>;
 }
+
+declare module "@sapper/common" {
+	export interface PreloadContext {
+		fetch: (url: string, options?: any) => Promise<any>;
+		error: (statusCode: number, message: Error | string) => void;
+		redirect: (statusCode: number, location: string) => void;
+	}
+
+	export interface PreloadPage {
```

Comment:
I wonder if we should call this just `Page`?

Reply:
I thought about this, too, and decided against it since `Page` is such a generic word that I fear it might clash with other people's namings. But I totally don't mind changing this to `Page`.

Reply:
Note that this definition already exists as `Page` in `runtime/src/app/types.ts`